lib/msf/core/exploit/remote/kerberos/client.rb

```diff
@@ -122,11 +122,106 @@ def send_request_as(opts = {})
           # @see Rex::Proto::Kerberos::Model::KdcResponse
           def send_request_tgs(opts = {})
             connect(opts)
-            req = build_tgs_request(opts)
+            req = opts.fetch(:req) { build_tgs_request(opts) }
             res = client.send_recv(req)
             disconnect
             res
           end
+
+          # Sends the required kerberos AS requests for a kerberos Ticket Granting Ticket
+          #
+          # @param options [Hash]
+          # @return [Msf::Exploit::Remote::Kerberos::Model::Tgt] ticket granting ticket
+          # @raise [Rex::Proto::Kerberos::Model::Error::KerberosError] if the provided credentials are invalid
+          def send_request_tgt(options = {})
+            realm = options[:realm]
+            server_name = options[:server_name]
+            client_name = options[:client_name]
+            password = options[:password]
+            request_pac = options.fetch(:request_pac, true)
+
+            # NOTE: This should be improved to properly negotiate encryption with the server, with graceful fallbacks etc. For now RC4_HMAC is hardcoded.
+            desired_encryption_type = Rex::Proto::Kerberos::Crypto::RC4_HMAC
+
+            unicode_password = Rex::Text.to_unicode(password)
+            password_digest = OpenSSL::Digest.digest('MD4', unicode_password)
```

Comment:
not needed right now but in the future maybe the user can just provide the password digest directly, so f they grabbed the keytab file they wouldn't need the password

Reply:
We should be able to support this in the future, i.e. by providing the keytab file directly - instead of getting folk to supply raw bytes themselves